File: packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/streamhandler/OnTransactionResultListener.java
/*
 * Copyright 2022, the Chromium project authors.  Please see the AUTHORS file
 * for details. All rights reserved. Use of this source code is governed by a
 * BSD-style license that can be found in the LICENSE file.
 */

package io.flutter.plugins.firebase.firestore.streamhandler;

import io.flutter.plugins.firebase.firestore.GeneratedAndroidFirebaseFirestore;
import java.util.List;

/** callback when a transaction result has been computed. */
public interface OnTransactionResultListener {
  void receiveTransactionResponse(
      GeneratedAndroidFirebaseFirestore.PigeonTransactionResult resultType,
      List<GeneratedAndroidFirebaseFirestore.PigeonTransactionCommand> commands);
}

File: packages/cloud_firestore/cloud_firestore/android/src/main/kotlin/io/flutter/plugins/firebase/firestore/streamhandler/OnTransactionResultListener.kt
/*
 * Copyright 2022, the Chromium project authors.  Please see the AUTHORS file
 * for details. All rights reserved. Use of this source code is governed by a
 * BSD-style license that can be found in the LICENSE file.
 */

package io.flutter.plugins.firebase.firestore.streamhandler

import io.flutter.plugins.firebase.firestore.InternalTransactionCommand
import io.flutter.plugins.firebase.firestore.InternalTransactionResult

/** Callback when a transaction result has been computed. */
fun interface OnTransactionResultListener {
  fun receiveTransactionResponse(
      resultType: InternalTransactionResult,
      commands: List<InternalTransactionCommand?>?
  )
}
